src/dbsync/include/dbsync.h

```diff
@@ -37,111 +30,211 @@
- * @param json_raw_snapshot JSON with snapshot values.
- * @param callback function callback to return individually result.
+ * @param handle    Handle assigned as part of the \ref dbsync_create method().
+ * @param js_insert JSON information with snapshot values.
  *
- * @return return 0 if the operations is success, otherwise an error code will vary depending on the operating system.
+ * @return 0 if succeeded,
+ *         specific error code (OS dependent) otherwise.
  */
-  EXPORTED int dbsync_update_with_snapshot_cb(
-    const DBSYNC_HANDLE handle,
-    const cJSON* json_snapshot,
-    void* callback);
+EXPORTED int dbsync_insert_data(const DBSYNC_HANDLE handle,
+                                const cJSON*        js_insert);
 
 /**
- * @brief Select data from table.
+ * @brief Sets the max rows in the \ref table table.
  *
- * @param handle Handle assigned and returned in the initialize method.
- * @param json_raw_data_input JSON with table name, fields to be return, and filters to apply in the query.
- * @param json_return_rows JSON that containt rows.
+ * @param handle   Handle assigned as part of the \ref dbsync_create method().
+ * @param table    Table name to apply the max rows configuration.
+ * @param max_rows Max rows number to be applied in the table \ref table table.
  *
- * @return return 0 if the operations is success, otherwise an error code will vary depending on the operating system.
+ * @return 0 if succeeded,
+ *         specific error code (OS dependent) otherwise.
+ *
+ * @details The table will work as a queue if the limit is exceeded.
  */
-  EXPORTED int dbsync_select_rows(
-    const DBSYNC_HANDLE handle,
-    const cJSON* json_data_input,
-    cJSON** json_return_rows);
+EXPORTED int dbsync_set_table_max_rows(const DBSYNC_HANDLE      handle,
+                                       const char*              table,
+                                       const unsigned long long max_rows);
 
 /**
- * @brief Set max rows in table, the table works as queue if the limit is exceeded.
+ * @brief Inserts (or modifies) a database record.
  *
- * @param handle Handle assigned and returned in the initialize method.
- * @param table Table name to apply the max rows configuration.
- * @param max_rows Quantity of max rows to apply in table.
+ * @param handle    Handle instance assigned as part of the \ref dbsync_create method().
+ * @param input     JSON information used to add/modified a database record.
+ * @param callback  Associated function callback with the results.
  *
- * @return return 0 if the operations is success, otherwise an error code will vary depending on the operating system.
+ * @return 0 if succeeded,
+ *         specific error code (OS dependent) otherwise.
  */
-  EXPORTED int dbsync_set_max_rows(
-    const DBSYNC_HANDLE handle,
-    const char* table,
-    const unsigned long long max_rows);
+EXPORTED int dbsync_sync_row(const DBSYNC_HANDLE handle,
```

Comment:
This function is ok. I think you are referring to the other existent one:
/**
 * @brief Synchronizes the \p js_input data using the \p txn current
 *  database transaction.
 *
 * @param txn      Database transaction to be used for \ref js_input data sync.
 * @param js_input JSON information to be synchronized.
 *
 * @return 0 if succeeded,
 *         specific error code (OS dependent) otherwise.
 */
EXPORTED int dbsync_sync_txn_row(const TXN_HANDLE txn,
                                 const cJSON*     js_input);